sync15-adapter/src/bso_record.rs

```diff
@@ -117,21 +131,97 @@ impl<T> DerefMut for BsoRecord<T> {
     }
 }
 
-impl<T> BsoRecord<T> {
-    /// If T is a Sync15Record, then you can/should just use record.into() instead!
+/// Represents the decrypted payload in a Bso. Provides a minimal layer of type safety to avoid double-encrypting.
+///
+/// Note: If we implement a full sync client in rust we may want to consider using stronger types for each record
```

Comment:
There's definitely some tension here between dealing with free-form collection names and arbitrary payloads, even though, in practice, the number of collections is fixed. I don't know if we'll be able to get away with not teaching the library anything about the different record types...or even if that's something we want. I think the way you're handling it now makes sense, though.

Reply:
I have part of a patch for managing collection names more sanely and efficiently, but it wasn't relevant to this. After stuff lands I'll pr it.